test/spec/add-member.test.js

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const test = require('tape')
+const { SecretKey } = require('ssb-private-group-keys')
+const { isValid } = require('../../spec/add-member')
+const GroupId = () => `%${new SecretKey().toString()}.cloaked`
+const FeedId = () => '@YXkE3TikkY4GFMX3lzXUllRkNTbj5E+604AkaO1xbz8=.ed25519'
+
+const Mock = (overwrite = {}) => {
+  const base = {
+    type: 'group/add-member',
+    version: 'v1',
+    groupKey: '3YUat1ylIUVGaCjotAvof09DhyFxE8iGbF6QxLlCWWc=',
+    root: '%THxjTGPuXvvxnbnAV7xVuVXdhDcmoNtDDN0j3UTxcd8=.sha256',
+    text: 'welcome keks!', // optional
+    recps: [
+      GroupId(),
+      // add 15 feedId
+    ],
+
+    tangles: {
+      group: {
+        root: '%THxjTGPuXvvxnbnAV7xVuVXdhDcmoNtDDN0j3UTxcd8=.sha256',
+        previous: ['%Sp294oBk7OJxizvPOlm6Sqk3fFJA2EQFiyJ1MS/BZ9E=.sha256'],
+      },
+      members: {
+        root: '%THxjTGPuXvvxnbnAV7xVuVXdhDcmoNtDDN0j3UTxcd8=.sha256',
+        previous: [
+          '%lm6Sqk3fFJA2EQFiyJ1MSASDASDASDASDASDAS/BZ9E=.sha256',
+          '%Sp294oBk7OJxizvPOlm6Sqk3fFJA2EQFiyJ1MS/BZ9E=.sha256',
+        ],
+      },
+    },
+  }
+  times(15, () => base.recps.push(FeedId()))
+
+  return Object.assign(base, overwrite)
+}
```

Comment:
I think these tests are good, but should live next to the schema in `private-group-spec`

Reply:
moved it there in this pr https://github.com/ssbc/private-group-spec/pull/19